block-streamer/src/bitmap.rs

```diff
@@ -0,0 +1,367 @@
+                while decompressed_bit_index + bit_index_offset
+                    >= (decompressed_byte_array.len() * 8)
+                {
+                    decompressed_byte_array.push(0b00000000);
+                }
+                self.set_bit(
+                    &mut decompressed_byte_array,
+                    decompressed_bit_index + bit_index_offset,
+                    true,
+                    true,
+                );
+                bit_index_offset = bit_index_offset + 1;
+            }
+
+            decompressed_bit_index += decoded_elias_gamma.value;
+            current_bit_value = !current_bit_value;
+        }
+
+        decompressed_byte_array
+    }
+
+    fn merge_compressed_bitmap_into_base_bitmap(
+        &self,
+        base_bitmap: &mut Bitmap,
+        compressed_bitmap: &Bitmap,
+    ) -> anyhow::Result<()> {
+        let decompressed_add_bitmap: Vec<u8> = self.decompress_bitmap(&compressed_bitmap.bitmap);
+        let start_bit_index: usize = match compressed_bitmap
+            .start_block_height
+            .checked_sub(base_bitmap.start_block_height)
+        {
+            Some(result) => result,
+            None => {
+                return Err(anyhow!(
+                    "Start block height in bitmap was lower than provided lowest block height",
+                ))
+            }
+        };
+
+        for bit_index_offset in 0..(decompressed_add_bitmap.len() * 8) {
+            let decompressed_bit_value = self.get_bit(&decompressed_add_bitmap, bit_index_offset);
+            while start_bit_index + bit_index_offset >= base_bitmap.bitmap.len() * 8 {
+                base_bitmap.bitmap.push(0b00000000);
+            }
+
+            self.set_bit(
+                &mut base_bitmap.bitmap,
+                start_bit_index + bit_index_offset,
+                decompressed_bit_value,
+                false,
+            );
+        }
+
+        Ok(())
+    }
+
+    pub fn get_merged_bitmap(
+        &self,
+        bitmaps_to_merge: &Vec<Base64Bitmap>,
+        smallest_start_block_height: usize,
+    ) -> anyhow::Result<Bitmap> {
+        let mut merged_bitmap: Bitmap = Bitmap {
+            bitmap: Vec::new(),
+            start_block_height: smallest_start_block_height,
+        };
+
+        for compressed_base64_bitmap in bitmaps_to_merge {
+            let decoded_bitmap: Vec<u8> =
+                general_purpose::STANDARD.decode(compressed_base64_bitmap.base64.clone())?;
+            let compressed_bitmap: Bitmap = Bitmap {
+                bitmap: decoded_bitmap,
+                start_block_height: compressed_base64_bitmap.start_block_height,
+            };
+            self.merge_compressed_bitmap_into_base_bitmap(&mut merged_bitmap, &compressed_bitmap)?;
+        }
+
+        Ok(merged_bitmap)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_getting_bit_from_array() {
```

Comment:
```suggestion
    fn getting_bit_from_array() {
```
Nit: `test_` seems superfluous here

Reply:
That's true. I'll reword the test names.